src/system/sys_profiler.cc

```diff
@@ -269,7 +375,15 @@ void Profiler::stop_timer(const CodePoint &cp) {
                 }
                 // close 'node' itself
                 timers_[actual_node].stop(false);
-                actual_node=timers_[actual_node].parent_timer;
+                actual_node = timers_[actual_node].parent_timer;
+                
+                // workaround for time-frame 0, if (actual_node >=0) would make more sense
+                //   but time-frame 0 has also parent_timer equal to 0
+                if (actual_node == child_timer && actual_node == 0)
```

Comment:
actual_node == child_timer should be enough. That's classical indication of the root of the rooted tree.